src/main/java/org/spongepowered/api/text/TextBuilder.java

```diff
@@ -0,0 +1,1124 @@
+ *
+ * Copyright (c) SpongePowered.org <http://www.spongepowered.org>
+ * Copyright (c) contributors
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package org.spongepowered.api.text;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+import com.google.common.base.MoreObjects;
+import com.google.common.base.Objects;
+import com.google.common.base.Optional;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.Iterables;
+import com.google.common.collect.Lists;
+import org.spongepowered.api.text.action.ClickAction;
+import org.spongepowered.api.text.action.HoverAction;
+import org.spongepowered.api.text.action.ShiftClickAction;
+import org.spongepowered.api.text.format.TextColor;
+import org.spongepowered.api.text.format.TextColors;
+import org.spongepowered.api.text.format.TextStyle;
+import org.spongepowered.api.text.format.TextStyles;
+import org.spongepowered.api.text.translation.Translation;
+
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+
+import javax.annotation.Nullable;
+
+/**
+ * Represents a builder class to create immutable {@link Text} instances.
+ *
+ * @see Text
+ */
+public abstract class TextBuilder {
+
+    protected TextColor color = TextColors.NONE;
+    protected TextStyle style = TextStyles.NONE;
+    protected List<Text> children = Lists.newArrayList();
+    @Nullable protected ClickAction<?> clickAction;
+    @Nullable protected HoverAction<?> hoverAction;
+    @Nullable protected ShiftClickAction<?> shiftClickAction;
+
+    /**
+     * Constructs a new empty {@link TextBuilder}.
+     */
+    protected TextBuilder() {
+    }
+
+    /**
+     * Constructs a new {@link TextBuilder} with the properties of the given
+     * {@link Text} as initial values.
+     *
+     * @param text The text to copy the values from
+     */
+    protected TextBuilder(Text text) {
```

Comment:
If "normally" there's two ways to build text (public everything-constructor and builder), then this should be public too